pyqtgraph/graphicsItems/PlotDataItem.py

```diff
@@ -790,58 +821,97 @@ def getData(self):
-                                # np.clip(self.yDisp, out=self.yDisp, a_min=min_val, a_max=max_val)
-                                fn.clip_array(self.yDisp, min_val, max_val, out=self.yDisp)
-                                self._drlLastClip = (min_val, max_val)
-                                # print('{:.1e}<->{:.1e}'.format( min_val, max_val ))
-                                x = self.xDisp
-                                y = self.yDisp
-                            else:
-                                # if none of the shortcuts worked, we need to recopy from the full data
-                                # print('alloc:', end='')
-                                # workaround for slowdown from numpy deprecation issues in 1.17 to 1.20+ :
-                                # y = np.clip(y, a_min=min_val, a_max=max_val)
-                                y = fn.clip_array(y, min_val, max_val)
-                                self._drlLastClip = (min_val, max_val)
-                                # print('{:.1e}<->{:.1e}'.format( min_val, max_val ))
+                            # print('alloc:', end='')
+                            # workaround for slowdown from numpy deprecation issues in 1.17 to 1.20+ :
+                            # y = np.clip(y, a_min=min_val, a_max=max_val)
+                            y = fn.clip_array(y, min_val, max_val)
+                            self._drlLastClip = (min_val, max_val)
+                            # print('{:.1e}<->{:.1e}'.format( min_val, max_val ))
         self.xDisp = x
         self.yDisp = y
         self.setProperty('xViewRangeWasChanged', False)
         self.setProperty('yViewRangeWasChanged', False)
+        # print('xDisp:', self.xDisp )
+        # print('xData:', self.xData )
+        # print('xMap.:', self.xMapped )
         return self.xDisp, self.yDisp
 
+    def _prepare_dataRect(self, x, y):
+        ### find bounds of plotable data ###
+        if y is None or x is None:
+            return None
+        selection = np.isfinite(y) # We are looking for the bounds of the plottable data set. Infinite and Nan are ignored. 
+        try:
+            ymin = np.min( y[selection] ) # find minimum of all finite values
+            ymax = np.max( y[selection] ) # find maximum of all finite values
+        except ValueError: # is raised then there are no finite values
+            ymin = np.nan
+            ymax = np.nan
+        selection = np.isfinite(x) # We are looking for the bounds of the plottable data set. Infinite and Nan are ignored. 
+        try:
+            xmin = np.min( x[selection] ) # find minimum of all finite values
+            xmax = np.max( x[selection] ) # find maximum of all finite values
+        except ValueError: # is reaised then there are no finite values
+            xmin = np.nan
+            xmax = np.nan
+        return QtCore.QRectF(
+            QtCore.QPointF(xmin,ymin),
+            QtCore.QPointF(xmax,ymax) )
```

Comment:
I was just testing this in the interpreter, thinking that constructing a QRectF with QPoints that have `np.nan` values would blow up, but it seems to work. 👍🏻